sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -49,11 +50,19 @@ import org.apache.spark.util.random.RandomSampler
  * The AstBuilder converts an ANTLR4 ParseTree into a catalyst Expression, LogicalPlan or
  * TableIdentifier.
  */
-class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging {
+class AstBuilder(conf: SQLConf)
+  extends SqlBaseBaseVisitor[AnyRef] with Logging with LookupCatalog {
   import ParserUtils._
 
   def this() = this(new SQLConf())
 
+  val defaultCatalog = for {
```

Comment:
It is not obvious what this is doing. I prefer using `map` and `flatMap` for readability.

Reply:
removed.